components/card/card_body.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import classNames from 'classnames';
+
+import './card.scss';
+
+type Props = {
+    expanded?: boolean;
+}
+
+export default class CardBody extends React.PureComponent<Props> {
+    card: React.RefObject<HTMLDivElement>;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.card = React.createRef();
+    }
+
+    componentDidMount() {
+        if (this.card.current && this.props.expanded) {
+            this.card.current.style.height = `${this.card.current.scrollHeight}px`;
+        }
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (this.card.current) {
```

Comment:
nit: Could reduce nesting by:

```javascript
if (!this.card.current) {
    return;
}

// do other stuff
```

Reply:
I wanted to have it nested in case there were other statements to add to `componentDidUpdate`. But since there aren't right now, I'll make this change.